File: Data Structures & Algorithms/is-anagram/submission-1.js
class Solution {
    /**
     * @param {string} s
     * @param {string} t
     * @return {boolean}
     */
    isAnagram(s, t) {
        const letters = {}
        for (let letter of s){
            if (letter in letters){
                letters[letter] += 1
            } else {
                letters[letter] = 1
            }
            
        }
        for (let letter of t){
            if (!(letter in letters)){

                return false
            } else {
                letters[letter] -= 1
                if (letters[letter] == -1){
                    return false
                }
            }
        }
        for (const key in letters){
            if (letters[key] != 0){
                return false
            }
        }
        return true
    }
}

File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-1.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        letters = {}
        L = 0
        output = 0
        for R  in range(len(s)):
            letters[s[R]] = letters.get(s[R], 0) + 1
            most_common = max(letters.values())

            if not most_common + k >= R-L+1:
                while max(letters.values()) + k < R - L + 1:
                    letters[s[L]] -= 1
                    L += 1
            output = max(output, R-L+1)
        return output